pimd/pim_mroute.h

```diff
@@ -163,19 +159,178 @@ struct igmpmsg {
 
 typedef struct mfcctl pim_mfcctl;
 
+typedef struct vifctl pim_vifctl;
+
+#define vc_vifi vifc_vifi
+#define vc_flags vifc_flags
+#define vc_threshold vifc_threshold
+#define vc_rate_limit vifc_rate_limit
+#define vc_lcl_addr vifc_lcl_addr
+#define vc_lcl_ifindex vifc_lcl_ifindex
+#define vc_rmt_addr vifc_rmt_addr
+
+#ifndef IGMPMSG_WRVIFWHOLE
+#define IGMPMSG_WRVIFWHOLE 4 /* For PIM processing */
+#endif
+
+#ifndef PIM_IPPROTO
+#define PIM_IPPROTO IPPROTO_IP
+#endif
+
 #else /* PIM_IPV != 4 */
+
+#include <netinet/ip6.h>
+#ifdef HAVE_NETINET6_IP6_MROUTE_H
+#include <netinet6/ip6_mroute.h>
+#endif
+
+#define IPV6_HEADER_LENGTH 40
+
 #if defined(HAVE_LINUX_MROUTE6_H)
 #include <linux/mroute6.h>
+#else
+
+
+#ifndef SIOCGETMIFCNT_IN6
+#define SIOCGETMIFCNT_IN6 SIOCPROTOPRIVATE /* IP protocol privates */
+#define SIOCGETSGCNT_IN6 (SIOCPROTOPRIVATE + 1)
+#define SIOCGETRPF (SIOCPROTOPRIVATE + 2)
+#endif
+
+#ifndef MRT6_INIT
```

Comment:
Rather than copying all these definitions from `<linux/mroute6.h>`, if needed we can provide a copy of `mroute6.h` in `include/linux` (but this should only be done if it's necessary due to larger changes in recent kernel versions)